wrappers/kmip/kmip.go

```diff
@@ -0,0 +1,537 @@
+// Copyright (c) 2025 OpenBao a Series of LF Projects, LLC
+// SPDX-License-Identifier: MPL-2.0
+
+// Initially inspired and adapted from ceph's internal kmip package
+// https://github.com/ceph/ceph-csi/blob/devel/internal/kms/kmip.go
+
+package kmip
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+
+	uuid "github.com/hashicorp/go-uuid"
+	wrapping "github.com/openbao/go-kms-wrapping/v2"
+	"github.com/ovh/kmip-go"
+	"github.com/ovh/kmip-go/kmipclient"
+	"github.com/ovh/kmip-go/ttlv"
+)
+
+// These constants contain the accepted env vars; the Vault one is for backwards compat
+const (
+	EnvKmipWrapperKeyId   = "KMIP_WRAPPER_KEY_ID"
+	EnvVaultKmipSealKeyId = "VAULT_KMIP_SEAL_KEY_ID"
+
+	EnvKmipEndpoint     = "KMIP_ENDPOINT"
```

Comment:
Do you mind using `BAO_` prefixed variables, unless these are understood by the `go-kmip` library? 

https://github.com/openbao/go-kms-wrapping/blob/9366f6b7c9ce05e7023a368584a6d315d7ba8040/wrappers/pkcs11/pkcs11_client.go#L69-L76

https://github.com/openbao/go-kms-wrapping/blob/9366f6b7c9ce05e7023a368584a6d315d7ba8040/wrappers/pkcs11/pkcs11_client.go#L95

Reply:
Done